src/pages/travel/visa/index.tsx

```diff
@@ -75,8 +274,67 @@ const VisaPage: React.FC = () => {
   >(new Map());
   const [dialogOpen, setDialogOpen] = useState(false);
   const [dialogCountry, setDialogCountry] = useState<string | null>(null);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [itemsPerPage] = useState(24); // 4x6 grid
   const searchInputRef = useRef<HTMLInputElement>(null);
 
+  // Pagination calculations
+  const totalPages = Math.ceil(filteredCountries.length / itemsPerPage);
+  const startIndex = (currentPage - 1) * itemsPerPage;
+  const endIndex = startIndex + itemsPerPage;
+  const currentCountries = filteredCountries.slice(startIndex, endIndex);
+
+  // Lazy loading flag component
+  const LazyFlag = React.memo(
+    ({ country, iso2 }: { country: string; iso2: string | undefined }) => {
+      const [isLoaded, setIsLoaded] = useState(false);
+      const [isInView, setIsInView] = useState(false);
+      const imgRef = useRef<HTMLImageElement>(null);
+
+      useEffect(() => {
+        const observer = new IntersectionObserver(
+          ([entry]) => {
+            if (entry.isIntersecting) {
+              setIsInView(true);
+              observer.disconnect();
+            }
+          },
+          { threshold: 0.1 }
+        );
+
+        if (imgRef.current) {
+          observer.observe(imgRef.current);
+        }
+
+        return () => observer.disconnect();
+      }, []);
+
+      return (
+        <div ref={imgRef} className='relative w-full h-full'>
+          {isInView && iso2 ? (
+            <Flag
+              code={iso2}
+              title={country}
+              alt={country}
+              className={`block w-full h-full object-cover transform -translate-x-6 opacity-0 transition-all duration-700 ease-[cubic-bezier(.22,.61,.36,1)] delay-75 group-hover:translate-x-0 group-hover:opacity-100 group-hover:scale-[1.02] will-change-transform motion-reduce:transition-none motion-reduce:transform-none motion-reduce:opacity-100 ${
+                isLoaded ? 'opacity-100' : 'opacity-0'
+              }`}
+              onLoad={() => setIsLoaded(true)}
```

Comment:
```suggestion
              loading='lazy'
```

This will simplify the on demand loading of images

Reply:
I'm not sure what effect you are referring to. I removed a lot of code and still have this animation.
![Image](https://github.com/user-attachments/assets/dc9e1f15-ba90-44e8-81d0-1b6134d96736)